Terra fast

faddat opened the pull request:
Hi there,

I've been working on integrating Terra's changes with mainline tm-db.

I think that you're doing great, praiseworthy work to improve database performance on the tendermint/cosmos stack.  

This PR includes some notable work that you may want to be aware of:

* a .gitpod.yml file that allows you to have a one-click dev env supporting rocksdb and cleveldb
* a test suite that runs against live gaia, which you could adapt to terra, or drop into terra's ci system.  If you want to use this test, you may consider using buildjet to run it. 
* changes to tests that should allow the use of the test suite, despite your changes to goleveldb and rocksdb

Given that work here is happening at a higher cadence, I would also like to suggest that you enable both issues and github actions on this repository.  

Thanks,
-Jacob

orthory committed:
fix: remove mutex in prefixdb

faddat committed:
Delete .gitpod.yml

faddat committed:
fmt

faddat committed:
fmt

faddat committed:
Delete .gitpod.yml

faddat committed:
remove mutex entirely instead of commenting out

faddat committed:
fmt

faddat committed:
remove .gitpod.yml

faddat committed:
update changelog

faddat committed:
Merge branch 'master' into remove-mutex

faddat committed:
Merge branch 'master' into grocksdb-and-docker

faddat committed:
Update Dockerfile

faddat committed:
Update README.md

trigger ci because codecov

faddat committed:
Update lint.yml

faddat committed:
Update lint.yml

always check lint

faddat committed:
prefixdb test

vuong177 committed:
all test passed in prefixdb_test.go

catShaark committed:
Merge branch 'master' into remove-mutex

vuong177 committed:
simple taskKey in prefixdb_test.go

faddat committed:
Merge branch 'master' into faddat/update-badger

faddat committed:
Update rocksdb.go

faddat committed:
Delete settings.json

faddat committed:
Update badger_db.go

faddat committed:
Update rocksdb.go

faddat committed:
Merge branch 'master' into grocksdb-and-docker

faddat committed:
gofumpt

faddat committed:
Update prefixdb.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

faddat committed:
Merge branch 'master' into remove-mutex

faddat committed:
Merge branch 'master' into patch-3

faddat committed:
Revert "gofumpt"

This reverts commit 8f6316c91b582c41443d8fb1999985f58cc06462.

faddat committed:
Merge branch 'master' into faddat/update-badger

faddat committed:
Merge branch 'master' into grocksdb-and-docker

faddat committed:
Merge branch 'master' into terra-fast

faddat committed:
fumpt

faddat committed:
Merge remote-tracking branch 'notional/fumpt' into HEAD

faddat committed:
Merge branch 'faddat/update-badger' into terra-fast

faddat committed:
Merge remote-tracking branch 'notional/faddat/update-badger' into terra-fast

faddat committed:
Merge remote-tracking branch 'fork/patch-3' into terra-fast

faddat committed:
fumpt

faddat committed:
fmt db test so pr passes linter

faddat committed:
Delete tools directory

using ghcr.io/faddat/cosmos allows us to not need a dockerfile separately for this repo

faddat committed:
Delete docker.yml

faddat committed:
Update ci.yml

faddat committed:
tidy

faddat committed:
bump grocksdb

faddat committed:
merge

faddat committed:
grocksdb

faddat committed:
add gitpod

faddat committed:
add cache to Rocksdb struct

faddat committed:
Update db_test.go

faddat committed:
no longer test for the redundant check

faddat committed:
db test fmt

faddat committed:
change tests

faddat committed:
fix for the removal of redundant checks

faddat committed:
now?

faddat committed:
don't declare itr

vuong177 committed:
refactor func Run

vuong177 committed:
check nil err before create PrefixDB in TestWithGolevelDB

catShaark committed:
add other db tests

faddat committed:
Merge branch 'master' into patch-3

faddat committed:
Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

faddat committed:
Merge branch 'master' into remove-mutex

catShaark committed:
update prefixdb_test.go

vuong177 committed:
minor defer

faddat committed:
Create gaia.yml

faddat committed:
Update gaia.yml

faddat committed:
Update gaia.yml

faddat committed:
Update cleveldb.go

faddat committed:
update test

faddat committed:
Merge branch 'terra-fast' of https://github.com/notional-labs/tm-db into terra-fast

faddat committed:
make a folder with cleveldb when needed

faddat committed:
Update gaia.yml

faddat committed:
add perf info and create snapshot folder individually with boltdb

faddat committed:
Merge branch 'terra-fast' of https://github.com/notional-labs/tm-db into terra-fast

faddat committed:
make folder the janky way

faddat committed:
use correct var name

faddat committed:
stutter: bbolt

faddat committed:
remove os and opts

faddat committed:
remove errors

faddat committed:
fix bolt

faddat committed:
fix bolt

faddat committed:
fix bolt

faddat committed:
fix bolt

hanjukim committed:
add DefaultOpenFilesCapacity for defining max open files

hanjukim committed:
use same open files capacity for every GOOS

hanjukim committed:
increase default max open file capacity to 32768 and use 4096 for mini
tag

faddat committed:
Merge branch 'master' into remove-mutex

faddat committed:
fixup with build tags

faddat committed:
Merge branch 'master' into patch-3

faddat committed:
Create .gitpod.yml

faddat committed:
fix linting

faddat committed:
itr.isInvalid would be unconditionally set anyhow, which is why the itr.isInvalid were considered to be ineffective assignments by the linter.

faddat committed:
Merge branch 'master' into terra-fast

faddat committed:
restore docker ci job

faddat committed:
Merge branch 'master' into remove-mutex

faddat committed:
Merge branch 'master' into patch-3

faddat committed:
Update cleveldb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

faddat committed:
Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

faddat committed:
Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

faddat committed:
Merge branch 'master' into terra-fast

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.46.2 to 1.47.0 (#259)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.46.2 to 1.47.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.46.2...v1.47.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

faddat committed:
Merge branch 'master' into terra-fast

faddat committed:
Merge branch 'master' into remove-mutex

faddat committed:
Firing this into CI, let's see.

faddat committed:
add filepath lib to tests

faddat committed:
fix filepath in goleveldb

faddat committed:
gofumpt formatting

faddat committed:
Merge remote-tracking branch 'terra/terra' into terra-fast

faddat committed:
Merge remote-tracking branch 'origin/remove-mutex' into terra-fast

faddat committed:
make gaia test work without buildjet

faddat committed:
integrate latest terra changes

faddat committed:
remove test invalidated by terra changes

faddat committed:
remove invalidated test

faddat committed:
space between comment and text

faddat committed:
use go 1.18 when testing gaia

faddat committed:
remove test invalidated by terra

faddat committed:
gofumpt enforcement

faddat committed:
timeout for gaia test

faddat committed:
remove invalid test

faddat committed:
cosmos

faddat committed:
use current gaia release branch

faddat committed:
no deadcode

faddat committed:
Merge remote-tracking branch 'faddat/patch-3' into terra-fast

faddat committed:
10 mins for state sync

dependabot[bot] committed:
build(deps): bump github.com/google/btree from 1.0.1 to 1.1.1 (#262)

Bumps [github.com/google/btree](https://github.com/google/btree) from 1.0.1 to 1.1.1.
- [Release notes](https://github.com/google/btree/releases)
- [Commits](https://github.com/google/btree/compare/v1.0.1...v1.1.1)

---
updated-dependencies:
- dependency-name: github.com/google/btree
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.1 to 1.7.2 (#261)

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.1 to 1.7.2.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.1...v1.7.2)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

faddat committed:
Merge branch 'master' into terra-fast

dependabot[bot] committed:
build(deps): bump github.com/google/btree from 1.1.1 to 1.1.2 (#264)

Lorenmod commented:
Kindly Live chat cosmos Agent on live chat support  https://terra-supports.com/  if you have issue or concerns

faddat committed:
Update .gitpod.yml

faddat committed:
tidy

faddat committed:
Merge remote-tracking branch 'upstream/master' into terra-fast

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.2 to 1.7.4 (#266)

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.4 to 1.7.5 (#267)

faddat committed:
Merge remote-tracking branch 'upstream/master' into terra-fast

faddat committed:
disable gaia sync tests

faddat committed:
skip logging into docker hub unless on the master branch, so we can test builds

faddat committed:
don't build the docker image, because we use a totally new environment

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.47.0 to 1.48.0 (#271)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.47.0 to 1.48.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.47.0...v1.48.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.5 to 1.8.0 (#270)

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.5 to 1.8.0.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.5...v1.8.0)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

faddat committed:
use only databases that are viable

faddat committed:
Merge remote-tracking branch 'upstream/master' into terra-fast

faddat committed:
tidy and improve readme file

faddat committed:
no pedantic

faddat committed:
Merge branch 'no-pedantic-thanks' into terra-fast

faddat committed:
revert breakage

faddat committed:
regenerate certs

faddat committed:
Merge remote-tracking branch 'notional/fix-main' into terra-fast

faddat committed:
fix: remove mutex in prefixdb (#239)

* fix: remove mutex in prefixdb

* fmt

* fmt

* Delete .gitpod.yml

* remove mutex entirely instead of commenting out

* fmt

* remove .gitpod.yml

* update changelog

* prefixdb test

* all test passed in prefixdb_test.go

* simple taskKey in prefixdb_test.go

* Update prefixdb.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

* fmt db test so pr passes linter

* refactor func Run

* check nil err before create PrefixDB in TestWithGolevelDB

* add other db tests

* Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

* update prefixdb_test.go

* minor defer

* fixup with build tags

* Create .gitpod.yml

* fix linting

* itr.isInvalid would be unconditionally set anyhow, which is why the itr.isInvalid were considered to be ineffective assignments by the linter.

* Update cleveldb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

* Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

* Update prefixdb_test.go

Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>

* Firing this into CI, let's see.

* add filepath lib to tests

* fix filepath in goleveldb

* gofumpt formatting

Co-authored-by: jess <jesse@soob.co>
Co-authored-by: vuong <nguyenvuong1122000@gmail.com>
Co-authored-by: khanh <50263489+catShaark@users.noreply.github.com>
Co-authored-by: M. J. Fromberger <michael.j.fromberger@gmail.com>
Co-authored-by: romelukaku <gundamaster5@gmail.com>

faddat committed:
regenerate certs (#280)

Commit:
Remove @creachadair from the CODEOWNERS file. (#285)

faddat committed:
Gofumpt second pass (#277)

* gofumpt, go 1.18, revive

* use go 1.18 for docker image

* update linters

* change linting

* regenerate certs

* go 1.17

* Update Dockerfile

luohaha committed:
fix cleveldb Stat() (#273)

Co-authored-by: Marko <marbar3778@yahoo.com>

faddat committed:
no pedantic (#279)

* no pedantic

* regenerate certs

faddat committed:
fixup
